src/devices/CharacterLcd/LcdCharacterEncodingFactory.cs

```diff
@@ -0,0 +1,934 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Iot.Device.CharacterLcd
+{
+    /// <summary>
+    /// Factory for creating Encodings that support different cultures on different LCD Displays.
+    /// </summary>
+    public class LcdCharacterEncodingFactory
+    {
+        private static readonly Dictionary<char, byte> DefaultA00Map;
+        private static readonly Dictionary<char, byte> DefaultA02Map;
+        private static readonly Dictionary<char, byte> DefaultCustomMap;
+
+        static LcdCharacterEncodingFactory()
+        {
+            // Default map which is used for unknown ROMs
+            DefaultCustomMap = new Dictionary<char, byte>();
+            // Inserts ASCII characters ' ' to 'z', which are common to most known character sets
+            for (char c = ' '; c <= 'z'; c++)
+            {
+                DefaultCustomMap.Add(c, (byte)c);
+            }
+
+            // The character map A00 contains the most used european letters, some greek math symbola plus japanese letters.
+            // Compare with the HD44780 specification sheet, page 17
+            DefaultA00Map = new Dictionary<char, byte>();
+            // Inserts ASCII characters ' ' to 'z', which are common to most known character sets
+            for (char c = ' '; c <= 'z'; c++)
+            {
+                DefaultA00Map.Add(c, (byte)c);
```

Comment:
Can't we add ASCII characters for all maps using only one loop?

Reply:
Possible, but not sure this increases readability.

Reply:
done